backends/platform/ios7/ios7_video.mm

```diff
@@ -613,8 +633,22 @@ - (void)updateMainSurface {
 	// Unfortunately we have to update the whole texture every frame, since glTexSubImage2D is actually slower in all cases
 	// due to the iPhone internals having to convert the whole texture back from its internal format when used.
 	// In the future we could use several tiled textures instead.
-	glTexImage2D(GL_TEXTURE_2D, 0, GL_RGB, _videoContext.screenTexture.w, _videoContext.screenTexture.h, 0, GL_RGB, GL_UNSIGNED_SHORT_5_6_5, _videoContext.screenTexture.getPixels()); printOpenGLError();
-
+	if (_videoContext.screenTexture.format == Graphics::PixelFormat(4, 8, 8, 8, 8, 24, 16, 8, 0)) {
+		// ARGB8888 - we need to convert it to RGBA8888
+		// Allocate a pixel buffer with 32 bits per pixel
+		void *pixelBuffer = malloc(_videoContext.screenTexture.h * _videoContext.screenTexture.w * sizeof(uint32_t));
+		// Copy the pixels from the texture, we don't want to operate directly the texture
+		memcpy(pixelBuffer, _videoContext.screenTexture.getPixels(), _videoContext.screenTexture.h * _videoContext.screenTexture.w * sizeof(uint32_t));
+		[self convertARGB888ToRGBA8888:pixelBuffer];
+
+		glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, _videoContext.screenTexture.w, _videoContext.screenTexture.h, 0, GL_RGBA, GL_UNSIGNED_BYTE, pixelBuffer); printOpenGLError();
```

Comment:
The use of `GL_UNSIGNED_BYTE` with `GL_RGBA` means the color is RBGA in memory, but ABGR in logical order (since little endian). So in effect the conversion above is from RGBA8888 to ABGR8888. And this is indeed consistent with the map you use (you reverse the byte order).